app/controllers/likes_controller.rb

```diff
@@ -0,0 +1,18 @@
+class LikesController < ApplicationController
+  before_action :require_login
+  
+  def create
+    @user = current_user
+    @image = Image.find(params[:image_id])
+    @like = @image.likes.create
+    redirect_to [@image.gallery, @image]
```

Comment:
You can use `redirect_to :back` here such that any page showing an image can show the like form, and clicking it will keep the user on that page. It's nifty.